plugins/ksp/src/org/jetbrains/kotlin/ksp/processing/impl/ResolverImpl.kt

```diff
@@ -162,8 +162,8 @@ class ResolverImpl(
 
     private val DeclarationDescriptor.containingScope: LexicalScope
         get() {
+            findPsi()?.let { return resolveSession.declarationScopeProvider.getResolutionScopeForDeclaration(it) }
```

Comment:
Can you elaborate on context on this change? Like a case where the old logic is not working.

Reply:
If I understand [getResolutionScopeForDeclaration](https://github.com/android/kotlin/blob/master/compiler/frontend/src/org/jetbrains/kotlin/resolve/lazy/DeclarationScopeProviderImpl.java#L46) correctly, it finds the enclosing scope of the given declaration. I think that is the behavior we want?

Reply:
For example, `findPsi()` always returns null for top level declarations, because `containingDescriptor` points to a package.

Reply:
sounds good.